library/core/src/iter/traits/iterator.rs

```diff
@@ -1761,11 +1761,11 @@ pub trait Iterator {
     /// Basic usage:
     ///
     /// ```
-    /// let a = [1, 2, 3];
+    /// let a = vec![1, 2, 3];
```

Comment:
This does not need to be `vec!`.

```suggestion
    /// let a = [1, 2, 3];
```

Reply:
without `vec!` it does not compile:

```
error[E0283]: type annotations needed
    --> test.rs:55:10
     |
55   |         .partition(|&n| n % 2 == 0);
     |          ^^^^^^^^^ cannot infer type for struct `Vec<_>`
     |
     = note: multiple `impl`s satisfying `Vec<_>: Extend<&i32>` found in the `alloc` crate:
             - impl<'a, T, A> Extend<&'a T> for Vec<T, A>
               where T: 'a, T: std::marker::Copy, A: 'a, A: Allocator;
             - impl<T, A> Extend<T> for Vec<T, A>
               where A: Allocator;

```

Reply:
it's due to this issue 
https://doc.rust-lang.org/nightly/edition-guide/rust-2021/IntoIterator-for-arrays.html

Reply:
I don't think https://doc.rust-lang.org/nightly/edition-guide/rust-2021/IntoIterator-for-arrays.html is the cause. Running the code:

```rust
fn main() {
    let a = [1, 2, 3];

    let (even, odd) = a.into_iter().partition::<Vec<_>, _>(|n| n % 2 == 0);

    assert_eq!(even, vec![2]);
    assert_eq!(odd, vec![1, 3]);
}
```

yields a different error with 2018 than the one you posted. Code examples in the docs should be on Rust 2021, so we don't need to worry about 2018. What code is in your `test.rs` file?

Reply:
if we are not worrying about 2018, it fine without the `vec!`